packages/next/shared/lib/image-loader.ts

```diff
@@ -1,5 +1,13 @@
-// TODO: change "any" to actual type
-function defaultLoader({ config, src, width, quality }: any): string {
+import type { ImageConfigComplete } from './image-config'
+
+type DefaultLoaderData = {
```

Comment:
The type is already defined here: 

https://github.com/vercel/next.js/blob/274649cf48cbfdc6fec1205fd7d72debaa8ab588/packages/next/client/image.tsx#L47-L50

Perhaps it should be moved to `./image-config.ts` so it can be shared between this file and `image.tsx`

Reply:
The `ImageConfig` that is used in `image.tsx` is slightly different than the one in `image-loader.ts`, it has an extra property: `allSizes`. So the types would be slightly changed if I share `ImageLoaderPropsWithConfig` between the two files. It doesn't seem to be a big deal, but I prefer to create two different types for them. I'm gonna move `ImageLoaderProps` to `image-loader.ts` so it can be shared between the two files.